lib/main.coffee

```diff
@@ -49,14 +54,17 @@ module.exports =
         filePath = textEditor.getPath()
         command = @executablePath
         ruleset = @rulesets
-        if @projectRules
-          rulesetPath = helpers.find(filePath, 'ruleset.xml')
+        if /^[a-z0-9]+\.xml$/gi.test(@rulesets)
+          rulesetPath = helpers.find(filePath, @rulesets)
           ruleset = rulesetPath if rulesetPath?
         parameters = []
         parameters.push(filePath)
         parameters.push('text')
         parameters.push(ruleset)
-        return helpers.exec(command, parameters).then (output) ->
+        options = {}
+        projectDir = atom.project.relativizePath(filePath)[0]
+        options.cwd = projectDir or textEditor.getDirectoryPath()
```

Comment:
That's the assumption I've been operating under... there is one linter that uses private stuff right now that just isn't available in the public API (and has been fixed for v1.9.x already since that would have broken it :stuck_out_tongue:), the rest stay on the public API as far as I know.